api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java

```diff
@@ -213,9 +216,35 @@ private Obs saveObsNotDirty(Obs obs, String changeMessage) {
 	private Obs saveNewOrVoidedObs(Obs obs, String changeMessage) {
 		Obs ret = dao.saveObs(obs);
 		saveObsGroup(ret,changeMessage);
+		saveObsReferenceRange(obs);
 		return ret;
 	}
 
+	private void saveObsReferenceRange(Obs obs) {
+		Concept concept = obs.getConcept();
+		ConceptService conceptService = Context.getConceptService();
+		ObsService obsService = Context.getObsService();
+
+		if (concept != null) {
+
+			List<ConceptReferenceRange> conceptReferenceRanges = conceptService.getConceptReferenceRangesByConceptId(concept.getId());
+			
+			ObsReferenceRange obsRefRange = new ObsReferenceRange();
+			obsRefRange.setObs(obs);
+			
+			if (!conceptReferenceRanges.isEmpty()) {
+				ConceptReferenceRange conceptReferenceRange = conceptReferenceRanges.get(0);
```

Comment:
Why are you choosing the first one? Why not the second? Why not the third, etc?

Reply:
I am open to suggestions here. I just picked the first one because in most cases we expect only one item in the list.

Reply:
We actually expect more than one concept reference range. For instance, one can be for Males, another for Females, etc.

Reply:
Did you forget to address this?

Reply:
Let me work on this now, I had not seen your response yet.